src/zip_code_tree.cpp

```diff
@@ -1852,6 +1952,24 @@ void ZipCodeTree::distance_iterator::continue_snarl() {
     }
 }
 
+void ZipCodeTree::distance_iterator::use_saved_traversal() {
+    auto next_traversal = pending_traversals.top();
+    pending_traversals.pop();
+    // Restore state
+    stack_data = next_traversal.stack_data;
+    chain_numbers = next_traversal.chain_numbers;
+    index = next_traversal.index;
+    right_to_left = next_traversal.right_to_left;
+    state(next_traversal.current_state);
```

Comment:
This looks like the domain-specific language stuff from the old PDA code has well and truly escaped. Do we still want this to be `state()` for adopting a state or should it be `set_state()`, or just `state = new_state`?